### ISSUE TITLE
[Bug]The width of space will be changed after copy&paste.

### ISSUE THREAD

**Issue by SnowyEvening** [JasonStein/Notepads#308]
**Describe the bug**
The width of space will be changed after copy&paste, when use a mono font which is "WenQuanYi Micro Hei Mono".

Font: WenQuanYi Micro Hei Mono
Font size: 18
Line break: CRLF
Write encoding: UTF-8
Read encoding: UTF-8
Tab: insert four spaces

**To Reproduce**
Enter a few spaces before entering words in every lines, then copy them and paste.

**Expected behavior**
space width will not be shorted.

**Screenshots**
![image](https://user-images.githubusercontent.com/33176053/75628543-1be0b000-5c15-11ea-9193-f67badb28428.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 1909 18363.693
 - Version: v1.0.3.0

**Comment by JasonStein** [comment]
Can you try a different mono font? Is it only happening to "WenQuanYi Micro Hei Mono"?

**Comment by SnowyEvening** [comment]
> Can you try a different mono font? Is it only happening to "WenQuanYi Micro Hei Mono"?

Yes. it happens in all mono font. When I use "DejaVu Sans Mono", It will still appear.
But, if I close the file after bug appears and reopen this file, it will be correct.

**Comment by JasonStein** [comment]
> > Can you try a different mono font? Is it only happening to "WenQuanYi Micro Hei Mono"?
> 
> Yes. it happens in all mono font. When I use "DejaVu Sans Mono", It will still appear.
> But, if I close the file after bug appears and reopen this file, it will be correct.

What IME are you using when typing the space and while copy/pasting the lines?

**Comment by SnowyEvening** [comment]
> > > Can you try a different mono font? Is it only happening to "WenQuanYi Micro Hei Mono"?
> > 
> > 
> > Yes. it happens in all mono font. When I use "DejaVu Sans Mono", It will still appear.
> > But, if I close the file after bug appears and reopen this file, it will be correct.
> 
> What IME are you using when typing the space and while copy/pasting the lines?

My Chinese IME is Sogou, and English IME is the default of Windows 10.
It will be appear 

![image](https://user-images.githubusercontent.com/33176053/75747139-1f7c5000-5d57-11ea-95b8-41c7696408d8.png)

![image](https://user-images.githubusercontent.com/33176053/75747195-3e7ae200-5d57-11ea-8c33-f3747b416f3d.png)

**Comment by JasonStein** [comment]
Ok, now I can re-pro it:

![image](https://user-images.githubusercontent.com/1166162/75813337-a96eed80-5d44-11ea-8026-78d45f1a3326.png)


But my case is differnt..... the first line is having the issue. I will take a look into it.

**Comment by JasonStein** [comment]
This is only happening when ou have Chinese characters (including the "period" character) in the selection. Not sure why but it might be some underline issue happening inside RichEdit.

**Comment by SnowyEvening** [comment]
It's a bit different, and it happens to me in English, and the location is different.
You can try typing English with spaces in front of both lines, then copy and paste these two lines, and it may reproduce.

![temp](https://user-images.githubusercontent.com/33176053/75938825-a64e3b80-5ec3-11ea-90e3-68c1301fd5be.gif)


Thanks for confirming and looking forward to the fix!
I really like this software, and hope to make some optimizations for the purpose of code viewing, such as line number, automatic indentation...

**Comment by soumyamahunt** [comment]
This happens with Cascadia Code font too(in English characters).

**Comment by JasonStein** [comment]
This is a behavior of text rendering controlled by RichEdit and the width of space will change based on the type of the character. If it is a double-byte character (Like Chinese), the space will be wider. I am not sure how to go around this since the font you are using is not Mono (I mean Chinese chars and English chars are not equal width).
